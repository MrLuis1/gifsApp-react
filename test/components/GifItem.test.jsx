import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('GifItem Component', () => {
  const title = 'luffy';
  const url = 'https://luffy.com/png';
  
  test('Snapshot match test', () => {
    const { container } = render( <GifItem title={ title } url={ url } /> )
    expect( container ).toMatchSnapshot();
  });

  test('Render the image from the url', () => {
    render(<GifItem title={ title } url={ url } />);
    // screen.debug(); Nos permite obtener el marcado html, con el metodo screen podemos verificar diversas etiquetas html y sus atributos

    // expect( screen.getByRole('img').src ).toBe( url );
    // expect( screen.getByRole('img').alt ).toBe( title );

    // ! De esta forma que se hace abajo se puede simplificar en caso de que se requiera hacer pruebas a varios atributos del elemento
    const { src, alt } = screen.getByRole('img');
    expect( src ).toBe( url );
    expect( alt ).toBe( title );
  });

  test('show Title in the component', () => {
    render(<GifItem title={ title } url={ url } />);
    expect( screen.getByText( title ) ).toBeTruthy();
  })
})