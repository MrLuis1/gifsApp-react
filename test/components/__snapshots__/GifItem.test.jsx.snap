// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem Component Snapshot match test 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="luffy"
      src="https://luffy.com/png"
    />
    <p>
      luffy
    </p>
  </div>
</div>
`;
